### ISSUE TITLE
LICENSE Issue GPLv2 compatibility with GPLv3

### ISSUE THREAD

**Issue by napsternxg** [brendano/ark-tweet-nlp#37]
Hi Brendan,

We are using your library for a twitter application which we plan to release under GPLv3 however, we cannot release your code with with our GPLv3 as it doesn't specify that your version of software is licensed under GPLv2 and later versions. 

So if you can change your license to GPLv2 or later then it will be easier to use your code in GPLv3 released code. 

You can see this compatibility matrix to see that **GPLv2 or later** is compatible with GPLv3 but not **GPLv2**

I would look forward to your response.

**Comment by brendano** [comment]
To clarify -- I do not think we have "released" our system as "GPL v2".  Rather, I thought our code was GPL v2 based on my interpretation of the GPL.  All source code that we actually wrote was licensed as Apache.  You're arguing that the libraries bundled with it to not necessitate "GPL v2", but rather, would be compatible with "GPL v2 or later".  This sounds possible to me, though I do not know.

In commit 31a43c2ce7b7c39d3fd149ffdb2d0e807844e342 I am updating the LICENSE.txt file to clarify that I think the license may well indeed be "GPL v2 or later".  However, I am not an expert and this may not be correct, so it's your job to get it correct if it matters for you.

I am replicating your email to me below in case this helps understand the issue.  Can we continue discussion on this Github issue so it's recorded for everyone? Thanks.

-----

From: "Mishra, Shubhanshu" <...@illinois.edu>

Looking at your code the GPL dependencies include:

`1.       Stanford POS tagger (GPLv2 or later) http://nlp.stanford.edu/software/tagger.shtml Even the version which your code is using 2010-05-26

`2.       Trove 3.0 - GNU Library or Lesser General Public License version 2.0 (LGPLv2) http://sourceforge.net/projects/trove4j/

It would be great if you too can release your software under GPLv2 or later rather than just GPLv2 as the later doesn’t play well with GPLv3 release and the former is compatible with more licenses. I have already filed an issue at your github repo at https://github.com/brendano/ark-tweet-nlp/issues/37
 
A full compatibility matrix of licenses can be seen at: http://www.gnu.org/licenses/gpl-faq.html#AllCompatibility
